hazelcast/src/main/java/com/hazelcast/spi/properties/GroupProperty.java

```diff
@@ -808,6 +808,14 @@
     public static final HazelcastProperty BIND_SPOOFING_CHECKS =
             new HazelcastProperty("hazelcast.nio.tcp.spoofing.checks", false);
 
+    /**
+     * Controls whether the task scheduler removes tasks immediately upon cancellation.
+     * This is disabled by default, because it can cause severe delays on other operations. By default all cancelled
+     * tasks will eventually get removed by scheduler workers.
+     */
+    public static final HazelcastProperty TASK_SCHEDULER_REMOVE_ON_CANCEL_POLICY =
+            new HazelcastProperty("hazelcast.executionservice.taskscheduler.remove.oncancel.policy", false);
```

Comment:
I would not call this policy, since it's just a `boolean` value. Most of our policy named configs have an `enum` or different implementations as configurable value. This will also fix the naming break from `removeOnCancelPolicy` to `removeOnCancel` further down in the code path.

Reply:
:+1: